x-pack/plugins/maps/public/components/map/mb/view.js

```diff
@@ -6,154 +6,252 @@
+       const geometryUnchanged = _.isEqual(this.props.tooltipState.activeFeature.geometry, targetFeature.geometry);
+       if(propertiesUnchanged && geometryUnchanged) {
+         return;
+       }
+     }
+
+     const layer = this._getLayer(targetFeature.layer.id);
+     const formattedProperties = await layer.getPropertiesForTooltip(targetFeature.properties);
+
+     let popupAnchorLocation = [e.lngLat.lng, e.lngLat.lat]; // default popup location to mouse location
+     if (targetFeature.geometry.type === 'Point') {
+       const coordinates = targetFeature.geometry.coordinates.slice();
+
+       // Ensure that if the map is zoomed out such that multiple
+       // copies of the feature are visible, the popup appears
+       // over the copy being pointed to.
+       while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+         coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+       }
+
+       popupAnchorLocation = coordinates;
+     }
+
+     this.props.setTooltipState({
+       activeFeature: {
+         properties: targetFeature.properties,
+         geometry: targetFeature.geometry
+       },
+       formattedProperties: formattedProperties,
+       layerId: layer.getId(),
+       location: popupAnchorLocation
+     });
+
+   }, 100);
+
+
+   _getMbLayerIdsForTooltips() {
+     return this.props.layerList.reduce((mbLayerIds, layer) => {
+       return layer.canShowTooltip() ? mbLayerIds.concat(layer.getMbLayerIds()) : mbLayerIds;
+     }, []);
+   }
+
+   _getMapState() {
+     const zoom = this._mbMap.getZoom();
+     const mbCenter = this._mbMap.getCenter();
+     const mbBounds = this._mbMap.getBounds();
+     return {
+       zoom: _.round(zoom, ZOOM_PRECISION),
+       center: {
+         lon: _.round(mbCenter.lng, DECIMAL_DEGREES_PRECISION),
+         lat: _.round(mbCenter.lat, DECIMAL_DEGREES_PRECISION)
+       },
+       extent: {
+         minLon: _.round(mbBounds.getWest(), DECIMAL_DEGREES_PRECISION),
+         minLat: _.round(mbBounds.getSouth(), DECIMAL_DEGREES_PRECISION),
+         maxLon: _.round(mbBounds.getEast(), DECIMAL_DEGREES_PRECISION),
+         maxLat: _.round(mbBounds.getNorth(), DECIMAL_DEGREES_PRECISION)
+       }
+     };
+   }
+
+   componentDidMount() {
+     this._initializeMap();
+     this._isMounted = true;
+   }
+
+   componentWillUnmount() {
+     this._isMounted = false;
+     if (this._checker) {
+       this._checker.destroy();
+     }
+     if (this._mbMap) {
+       this._mbMap.remove();
+       this._mbMap.off('mousemove', this._onMouseMove);
+       this._mbMap = null;
+       this._tooltipContainer = null;
+     }
+     this.props.onMapDestroyed();
+   }
+
+   async _initializeMap() {
+
+
+     this._mbMap = await createMbMapInstance(this.refs.mapContainer, this.props.goto ? this.props.goto.center : null);
+
+     if (!this._isMounted) {
+       return;
+     }
+     // Override mapboxgl.Map "on" and "removeLayer" methods so we can track layer listeners
```

Comment:
We are no longer using layer level listeners. Do we still need this logic? I am ok with keeping it, it is just no longer being used but could be used in the future

Reply:
nice catch. yeah, for sure, this needs to go.